morpheus/utils/nvt/schema_converters.py

```diff
@@ -0,0 +1,430 @@
+        deps: [
+            MutateOp(lambda selector,
+                     df: df.rename(columns={ci.input_name: ci.name}),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[RenameColumn] '{ci.input_name}' => '{ci.name}'")
+        ],
+    StringCatColumn:
+        lambda ci,
+        deps: [
+            MutateOp(partial(nvt_string_cat_col, output_column=ci.name, input_columns=ci.input_columns, sep=ci.sep),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[StringCatColumn] '{','.join(ci.input_columns)}' => '{ci.name}'")
+        ],
+    StringJoinColumn:
+        lambda ci,
+        deps: [
+            MutateOp(partial(
+                nvt_string_cat_col, output_column=ci.name, input_columns=[ci.name, ci.input_name], sep=ci.sep),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[StringJoinColumn] '{ci.input_name}' => '{ci.name}'")
+        ],
+    JSONFlattenInfo:
+        lambda ci,
+        deps: [json_flatten_from_input_schema(ci.input_col_names, ci.output_col_names)]
+}
+
+
+def build_nx_dependency_graph(column_info_objects: typing.List[ColumnInfo]) -> nx.DiGraph:
+    graph = nx.DiGraph()
+
+    def find_dependent_column(name, current_name):
+        for ci in column_info_objects:
+            if ci.name == current_name:
+                continue
+            if ci.name == name:
+                return ci
+            elif ci.__class__ == JSONFlattenInfo:
+                if name in [c for c, _ in ci.output_col_names]:
+                    return ci
+        return None
+
+    for col_info in column_info_objects:
+        graph.add_node(col_info.name)
+
+        if col_info.__class__ in [RenameColumn, BoolColumn, DateTimeColumn, StringJoinColumn, IncrementColumn]:
+            # If col_info.name != col_info.input_name then we're creating a potential dependency
+            if col_info.name != col_info.input_name:
+                dep_col_info = find_dependent_column(col_info.input_name, col_info.name)
+                if dep_col_info:
+                    # This CI is dependent on the dep_col_info CI
+                    graph.add_edge(dep_col_info.name, col_info.name)
+
+        elif col_info.__class__ == StringCatColumn:
+            for input_col_name in col_info.input_columns:
+                dep_col_info = find_dependent_column(input_col_name, col_info.name)
+                if dep_col_info:
+                    graph.add_edge(dep_col_info.name, col_info.name)
+
+        elif col_info.__class__ == JSONFlattenInfo:
+            for output_col_name in [c for c, _ in col_info.output_col_names]:
+                dep_col_info = find_dependent_column(output_col_name, col_info.name)
+                if dep_col_info:
+                    graph.add_edge(dep_col_info.name, col_info.name)
```

Comment:
Can this functionality be baked into the ColumnInfo classes themselves? Like a virtual function that returns the dependent column info? This would help locate the functionality with the class

Reply:
I'll take a look at this; I'd originally tried to avoid modifying the ColumnInfo classes, but based on our discussions, it sounds like we might want to continue using them going forward and just translate to NVT.

Reply:
I want to spend more time thinking about the best way to do this. If we're going to use ColumnInfo to define workflows, lets save this for the next round of updates.